Makefile

```diff
@@ -82,11 +84,23 @@ $(GOLANGCI_LINT_BINARY):
 	# see https://github.com/golangci/golangci-lint
 	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(GO_PATH)/bin $(GOLANGCI_LINT_VERSION)
 
+# not PHONY, installs deps if it doesn't exist
+$(DEP_BINARY):
+	curl https://raw.githubusercontent.com/golang/dep/master/install.sh | DEP_RELEASE_TAG=$(DEP_RELEASE_TAG) sh
+
+.PHONY: ensure
```

Comment:
Instead of using the `ifeq` check, can't you use make target dependencies like this?
```
.PHONY: ensure
ensure: $(DEP_BINARY)
	$(DEP_BINARY) ensure
```

Reply:
Sure and fixed